Fixes #12257 - added Debian kexec template

lzap opened the pull request.

lzap committed:
Fixes #12257 - added Debian kexec template

stbenjam commented:
I get:

```
2015-10-22T10:41:25 [app] [W] Reloading kernel on mac5254002b5b69.example.com task failed with the following error: un
defined method `url_for_boot' for #<Debian:0x000000023cd618>
 | NoMethodError: undefined method `url_for_boot' for #<Debian:0x000000023cd618>
```

For whatever reason Debian OS's don't have this url_for_boot method? Everyone else seems to

lzap commented:
Sorry this took so long :-) @mmoll actually pointed out on the list how this is solved in iPXE template, I copied the trick over here:

```
boot_files_uris = @host.operatingsystem.boot_files_uri(@host.medium, @host.architecture)
```

https://github.com/theforeman/community-templates/blob/develop/preseed/iPXE.erb

Rebased, hopefully it works now.

mmoll commented on app/views/foreman_discovery/debian_kexec.erb:
```diff
@@ -0,0 +1,36 @@
+kind: kexec
+name: Debian kexec
+oses:
+oses:
```
duplicate line

mmoll commented on app/views/foreman_discovery/debian_kexec.erb:
```diff
@@ -0,0 +1,36 @@
+oses:
+oses:
+- Debian 6.0
+- Debian 7.
```
in the community-templates, we did cut that down to just Debian and Ubuntu to let all versions associate

lzap commented:
Develop branch is utterly broken right now, need to take a look.

lzap commented:
This can be now tested [test]

lzap commented:
Please ignore the failure, its javascript error.

dLobatog commented:
I'm going to test with Debian & Ubuntu & report now..

dLobatog commented:
Holding off because I found http://projects.theforeman.org/issues/15644 trying this out

lzap commented:
What's the status? Have you provided correct link above? I think this one is not related.

dLobatog commented:
@lzap Yup, sorry. I'll try this one tomorrow. The ticket I linked to wasn't caused by this, but I couldn't keep on testing without the fix because the discovery host facts couldn't be submitted without the patch.

lzap commented:
Fixed finally, the problem was `url_for_boot` was not available for Debian. Also, both @kernel and @initrd variables are overriden when doing template preview, this was very confusing. It is only set properly during provisioning of a discovered host. Therefore I changed the templates to set the fields properly themself, I am keeping the two attributes just for compatibility.

During testing I found two regressions, you need to merge them before testing this. I will link them both to this PR.

lzap commented:
Test this with Katello installed if possible as it overrides some mediapath etc.

lzap commented:
@dLobatog @stbenjam since this one is quite work to test, I advise you to review and merge the other three PRs with Important label and do this as the last one if possible. I can perhaps make a Discovery 6.0 release without this patch, but the others are more important. Thanks!

lzap commented:
Rebased, thanks for reviews @dLobatog !

dLobatog commented:
Success - please add to the release notes that it also works with Ubuntu (I tried with 14.04 & it worked). Great job, merging!

dLobatog commented:
Merged as e940d32c6e2958e3ab048b1d5c13689748d4dc9e, thanks @lzap!
